build-docker.sh

```diff
@@ -1,56 +1,82 @@
 #!/bin/bash
```

Comment:
Can we document this script saying what it does? I think we should also mentions
- it's for user purpose only
- they should check the CI script as well because that's the script actually tested in our CI pipeline

Reply:
Definitely!